### ISSUE TITLE
API 2.0 Support?

### ISSUE THREAD

**Issue by MCKLtech** [intercom/intercom-php#293]
Hey Intercom,

Is support for API 2.0 in the PHP SDK planned?

Thanks!

**Comment by giannidhooge** [comment]
Got a reply from Intercom Support:

> There's no plan as yet to remove support for the previous versions of the API.  So far, only the REST API and the Ruby SDK support version 2.0, but our team are working on rolling this out to the other supported SDKs also.  I don't have a timeline on when this will be completed, but keep an eye on the Github repo for updates.

**Comment by GabrielAnca** [comment]
Hi @MCKLtech & @giannidhooge 👋 As said, we don't have a timeline on this for the PHP SDK yet. However, if anybody wants to give it a shot and add support for version 2.0, feel free to jump in and leave a note here so we are all aligned and we don't do double work 😃

**Comment by GabrielAnca** [comment]
Just leaving a note here to note that @JakeBooher started working on adding the conversations and contacts search endpoints: https://github.com/intercom/intercom-php/pull/302 - just in case anybody wants to jump in and add suggestions! 😃

**Comment by GabrielAnca** [comment]
Hi! @lloydliyu made a new PR including support for both the contacts and conversations endpoints https://github.com/intercom/intercom-php/pull/306. If anybody wants to leave feedback we would be very happy!

**Comment by RaneLafraze** [comment]
@GabrielAnca I took a look at the PR #306 this morning and am hopeful that these changes will be finalized and pushed to production for use soon.

**Comment by robbieaverill** [comment]
Hey team, I'd encourage reviewing of open PRs that contribute to supporting the new APIs here. I've just installed this SDK and scratched my head for a good half hour before realising that the "Users" API has been removed from the version of the API that the SDK used under the hood. Instead, I'm using the "Contacts" API manually with `$client->post()` to search and `$client->delete()` to delete my users.

I assume that deleting a contact is the equivalent of the formerly noted "permanently delete user" action, required for GDPR compliance. The Contacts API doesn't mention permanent deletion, but there is a different archive API which implies it is permanent.

Anyway - installing this SDK out of the box and using it to manage users is pointless right now because it doesn't work at all.

**Comment by GabrielAnca** [comment]
This was closed when I merged the linked PR, but we are not ready to release a new version yet.

I made this PR: https://github.com/intercom/intercom-php/issues/293 which updates the readme with info about what's supported on each api version. I didn't have a chance to test it yet, but I'm hoping to get it tested today and releasing a version this week.

@robbieaverill thank you for your feedback 🙇 I hope the new version that will be release covers your needs. It would be great to know if you would be missing any other feature, so either us or any of the community users can also help getting that added to the SDK.

Regarding your assumption about permanent deletion, you are totally right. We aimed to separate permanent deletion and temporary deletion (archival) to make it easier to understand.
